scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,342 @@
+        if self.trace_io:
+            print(f"{SGR_TRACE}{topic}:{SGR_RESET} {message}")
+
+    def receive_message(self) -> Any:
+        # `, timeout: float = 2.0`
+        # Note, we should make use of timeout to avoid infinite blocking if nothing is received.
+        LEN_HEADER = "Content-Length: "
+        TYPE_HEADER = "Content-Type: "
+        if self.process.stdout == None:
+            return None
+        message_size = None
+        while True:
+            #read header
+            line = self.process.stdout.readline()
+            if not line:
+                # server quit
+                return None
+            line = line.decode("utf-8")
+            if not line.endswith("\r\n"):
+                raise RuntimeError("Bad header: missing newline")
+            # remove the "\r\n"
+            line = line[:-2]
+            if line == "":
+                break # done with the headers
+            if line.startswith(LEN_HEADER):
+                line = line[len(LEN_HEADER):]
+                if not line.isdigit():
+                    raise RuntimeError("Bad header: size is not int")
+                message_size = int(line)
+            elif line.startswith(TYPE_HEADER):
+                # nothing todo with type for now.
+                pass
+            else:
+                raise RuntimeError("Bad header: unknown header")
+        if not message_size:
+            raise RuntimeError("Bad header: missing size")
+        rpc_message = self.process.stdout.read(message_size).decode("utf-8")
+        self.trace('receive_message', rpc_message)
+        return json.loads(rpc_message)
+
+    def send_message(self, method_name: str, params: Any) -> None:
+        if self.process.stdin == None:
+            return
+        message = { 'jsonrpc': '2.0', 'method': method_name, 'params': params }
+        json_string = json.dumps(obj=message, cls=MyEncoder)
+        rpc_message = f"Content-Length: {len(json_string)}\r\n\r\n{json_string}"
+        self.trace(f'send_message ({method_name})', json_string)
+        self.process.stdin.write(rpc_message.encode())
+        self.process.stdin.flush()
+
+    def call_method(self, method_name: str, params: Any) -> Any:
+        self.send_message(method_name, params)
+        return self.receive_message()
+
+    def send_notification(self, name: str, params: Any = None) -> None:
+        self.send_message(name, params)
+
+# }}}
+
+SGR_RESET = '\033[m'
+SGR_TRACE = '\033[1;36m'
+SGR_TEST_BEGIN = '\033[1;33m'
+SGR_ASSERT_BEGIN = '\033[1;34m'
+SGR_STATUS_OKAY = '\033[1;32m'
+SGR_STATUS_FAIL = '\033[1;31m'
+
+class ExpectationFailed(Exception):
+    def __init__(self, actual, expected):
+        self.actual = actual
+        self.expected = expected
+        diff = DeepDiff(actual, expected)
+        super().__init__(
+            f"Expectation failed. Expected {expected} but got {actual}. {diff}"
+        )
+
+# pylint: disable-next=too-many-instance-attributes
+class SolidityLSPTestSuite: # {{{
```

Comment:
Python has a built-in unit testing library, does it makes sense to use it?
https://docs.python.org/3/library/unittest.html

Reply:
One more thing, if you decide not to roll your tests into `pyscriptTests.py`, don't forget to add them to `tests.sh` next to it:
https://github.com/ethereum/solidity/blob/dcef56a531b6181135aabebf019d7a916bdf6056/scripts/tests.sh#L83-L84

Reply:
@cameel i tried hard porting to python test. While it can be made, it took me way too long. I'll defer that to after merge/release if possible (think so) to "just finish with the missing tests". :)

Adding to tests.sh. Thx for the tip.

Reply:
Maybe you could create an issue to rewrite it in unittest? I really think we should not reinvent the wheel here but I agree that it can be done afterwards.